client/handle_http.go

```diff
@@ -848,16 +849,40 @@ func UploadFile(src string, origDest *url.URL, token string, namespace namespace
 		nonZeroSize = fileInfo.Size() > 0
 	}
 
-	// Parse the writeback host as a URL
-	writebackhostUrl, err := url.Parse(namespace.WriteBackHost)
+	// call a GET on the director, director will respond with our endpoint
+	directorUrlStr := param.Federation_DirectorUrl.GetString()
+	directorUrl, err := url.Parse(directorUrlStr)
 	if err != nil {
-		return 0, err
+		return 0, errors.Wrap(err, "failed to parse director url")
+	}
+	directorUrl.Path, err = url.JoinPath("/api/v1.0/director/origin", origDest.Path)
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to parse director path for upload")
+	}
+
+	payload := []byte("forPUT")
+	req, err := http.NewRequest("GET", directorUrl.String(), bytes.NewBuffer(payload))
```

Comment:
Is it a put? It's not doing an update to the director, is it? It's getting information.

Reply:
@bbockelm Just want to make sure I am approaching this correctly: 
So should I move this logic to around where we query the director/`CreateNsFromDirectorResp` within `client/main.go` and populate `writebackhost` with the correct value in cases we do not use topology. Then change `uploadfile` to get the destination from `writebackhost` in the namespace struct?

Reply:
One tweak:

> Then change uploadfile to get the destination from writebackhost in the namespace struct?

I'd say change uploadfile _back_ to this behavior.  Basically, revert the changes in that method back to where it was before.

Reply:
Oh, another clarification:
> move this logic to around where we query the director
The logic should also change slightly too.  Instead of a `GET` with body contents, it should be a `PUT`.